desk/app/activity.hoon

```diff
@@ -813,27 +755,20 @@
   ^-  activity-summary:a
   %-  (log |.("summarizing unreads for: {<source>}"))
   =/  top=time  -:(fall (ram:on-event:a stream.index) [*@da ~])
-  ::  for each item in reads
-  ::  omit:
-  ::    if we don't have unreads enabled for that event
-  ::    any items that are unread for some reason
-  ::  then remove the post or reply from the event stream
-  ::  and call stream-to-unreads
-  ::
-  ::  TODO: flip around and iterate over stream once, cleaning reads out
-  ::        and segment replies for unread threads tracking
-  =;  unread-stream=stream:a
-    =/  children  (get-children:src indices source)
-    %-  (log |.("children: {<?:(?=(%base -.source) 'all' children)>}"))
-    (stream-to-unreads source index(stream unread-stream) children top)
-  ?:  ?=(%base -.source)  ~
-  %+  gas:on-event:a  *stream:a
-  %+  murn
-    (tap:on-event:a (lot:on-event:a stream.index `floor.reads.index ~))
-  |=  [=time =event:a]
-  ?:  child.event  ~
-  ?:  (has:on-read-items:a items.reads.index time)  ~
-  `[time event]
+  =/  unread-stream=stream:a
+    ::  all base's events are from children so we can ignore
+    ?:  ?=(%base -.source)  ~
+    ::  we don't need to take child events into account when summarizing
+    ::  the activity, so we filter them out
+    =-  +:-
+    %^  (dip:on-event:a @)  stream.index  ~
+    |=  [st=@ =time-id:a =event:a]
+    :_  [%.n st]
+    ?.  &(!child.event (gth time-id floor.reads.index))  ~
+    `event
```

Comment:
We use a weird construction with `+dip` here because the alternative is tapping, murning and gassing, like the old code did, right? Is this new implementation measurably faster?

Previously we used `+lot` to limit the amount of items we processed. Do we want to early-out once we reach a `time-id` that's beyond the `floor.reads.index`?